Assignment1/irisdata.py

```diff
@@ -0,0 +1,246 @@
+import sys
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+
+
+def main():
+    # Loading the dataset
+    data_df = pd.read_csv("Iris.csv")
+    print(data_df.head())
+    print(data_df.tail())
+    # Getting simple summary stats using pandas
+    print(data_df.describe())
+    # checking the columns of data
+    print(data_df.columns)
+    # checking the count of species
+    print(data_df["species"].value_counts())
+    # converting the data to numpy array and getting summary
+    x = data_df.iloc[:, 0:4].values
+    # summary of the data
+    print(
+        "Sepal Length Summary [",
+        "Mean :",
+        np.mean(x[:, 0]),
+        "Median :",
+        np.median(x[:, 0], axis=None),
+        "Minimum:",
+        np.min(x[:, 0], axis=None),
+        "Maximum :",
+        np.max(x[:, 0], axis=None),
+        "1st quartile:",
+        np.quantile(x[:, 0], 0.25, axis=None),
+        "3rd Quartile",
+        np.quantile(x[:, 0], 0.75, axis=None),
+    )
+    print(
+        "Sepal Width Summary [",
+        "Mean :",
+        np.mean(x[:, 1]),
+        "Median :",
+        np.median(x[:, 1], axis=None),
+        "Minimum:",
+        np.min(x[:, 1], axis=None),
+        "Maximum :",
+        np.max(x[:, 1], axis=None),
+        "1st quartile:",
+        np.quantile(x[:, 1], 0.25, axis=None),
+        "3rd Quartile",
+        np.quantile(x[:, 1], 0.75, axis=None),
+    )
+    print(
+        "Petal Length Summary [",
+        "Mean :",
+        np.mean(x[:, 2]),
+        "Median :",
+        np.median(x[:, 2], axis=None),
+        "Minimum:",
+        np.min(x[:, 2], axis=None),
+        "Maximum :",
+        np.max(x[:, 2], axis=None),
+        "1st quartile:",
+        np.quantile(x[:, 2], 0.25, axis=None),
+        "3rd Quartile",
+        np.quantile(x[:, 2], 0.75, axis=None),
+    )
+    print(
+        "Petal Width Summary [",
+        "Mean :",
+        np.mean(x[:, 3]),
+        "Median :",
+        np.median(x[:, 3], axis=None),
+        "Minimum:",
+        np.min(x[:, 3], axis=None),
+        "Maximum :",
+        np.max(x[:, 3], axis=None),
+        "1st quartile:",
+        np.quantile(x[:, 3], 0.25, axis=None),
+        "3rd Quartile",
+        np.quantile(x[:, 3], 0.75, axis=None),
+    )
+    # scatter plots
+    # scatter plot of sepal length and sepal width
+    fig = px.scatter(
+        data_df,
+        x="sepal.width",
+        y="sepal.length",
+        color="species",
+        symbol="species",
+        title="Scatter plot of Seoal Length & Sepal Width",
+    )
+    fig.show()
```

Comment:
For the figures, you can save the output as HTML files so that when someone tries to run the code it does not open multiple tabs on their system.

Reply:
Yeah sure!